### PR TITLE
test coverage reports

### PR DESCRIPTION
**Pull request by martin-volf**
Test coverage can be measured now:
```
$ git clone https://github.com/NSO-developer/drned-xmnr/ -b mvolf/test-coverage
...
$ cd test
$ pip install -r unit/requirements.txt; pip install paramiko coverage
...
$ ./test-coverage
... (pytest and lux running)

Run "coverage report" or "coverage html" now
$ coverage report
Name                                                          Stmts   Miss  Cover
---------------------------------------------------------------------------------
/home/opt/drned-xmnr/drned-skeleton/cli2netconf.py                    80      8    90%
/home/opt/drned-xmnr/drned-skeleton/devcli.py                        201     27    87%
/home/opt/drned-xmnr/drned-skeleton/load-default-config.py            22     22     0%
/home/opt/drned-xmnr/drned-skeleton/save-default-config.py            15     15     0%
/home/opt/drned-xmnr/python/drned_xmnr/__init__.py                     0      0   100%
/home/opt/drned-xmnr/python/drned_xmnr/action.py                     100     29    71%
/home/opt/drned-xmnr/python/drned_xmnr/check_action.py                43      2    95%
/home/opt/drned-xmnr/python/drned_xmnr/op/__init__.py                  0      0   100%
/home/opt/drned-xmnr/python/drned_xmnr/op/base_op.py                 307     24    92%
/home/opt/drned-xmnr/python/drned_xmnr/op/common_op.py                67     26    61%
/home/opt/drned-xmnr/python/drned_xmnr/op/config_op.py               347     38    89%
/home/opt/drned-xmnr/python/drned_xmnr/op/coverage_op.py              67      4    94%
/home/opt/drned-xmnr/python/drned_xmnr/op/ex.py                        8      1    88%
/home/opt/drned-xmnr/python/drned_xmnr/op/filtering/__init__.py        3      0   100%
/home/opt/drned-xmnr/python/drned_xmnr/op/filtering/cort.py           21      0   100%
/home/opt/drned-xmnr/python/drned_xmnr/op/filtering/events.py        216     20    91%
/home/opt/drned-xmnr/python/drned_xmnr/op/filtering/filtering.py      29      0   100%
/home/opt/drned-xmnr/python/drned_xmnr/op/filtering/states.py        227      4    98%
/home/opt/drned-xmnr/python/drned_xmnr/op/setup_op.py                155     69    55%
/home/opt/drned-xmnr/python/drned_xmnr/op/transitions_op.py          206      3    99%
---------------------------------------------------------------------------------
TOTAL                                                          2114    292    86%
$ 
```
The change touches one the sources only to allow running coverage-instructed Python, almost all changes are in test test/ subtree.

### CODE REVIEW

File: test/covplugins.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import os
+import coverage
+
+
+def lookup_path(filename):
+    if 'namespaces' in filename:
```
[the remark this answers is not in the document]
Comment (martin-volf) [reply test/covplugins.py]: Changed to `'/namespaces/'`.

File: test/lux/common.luxinc [hunk 2]
```diff
@@ -80,6 +80,25 @@
     !rm $nsdir/aaa_init.xml
 [endmacro]
 
+[macro coverage]
```
[the remark this answers is not in the document]
Comment (martin-volf) [reply test/lux/common.luxinc]: Fair enough, it is `start-ncs-with/without-coverage` now.

File: test/lux/coverage-start [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+if [ -z "$XMNR_ROOT"]
+then
+    export XMNR_ROOT=$(readlink -f ../../..)
+fi
+pypath=${NCS_DIR}/src/ncs/pyapi:$XMNR_ROOT/test
+
+if [ "x$PYTHONPATH" != "x" ]; then
```
[the remark this answers is not in the document]
Comment (martin-volf) [reply test/lux/coverage-start]: Right, I blindly copied that from the default python-vm start script; I guess it is used to keep compatibility with older shells, but who cares about `sh`; fixed.

**Review by micnovak — APPROVED** [review]

File: test/lux/coverage-start [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
```
Comment (jojohans) [inline test/lux/coverage-start:1]: Does this path work if we run coverage on GitHub too?

**Review by jojohans — APPROVED** [review]

**Pull request closed without merging**